__tests__/jest_extensions.js

```diff
@@ -0,0 +1,145 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import process from 'node:process';
+import { fileURLToPath } from 'node:url';
+
+import { expect } from '@jest/globals';
+import { diff } from 'jest-diff';
+import pg from 'pg';
+import Papa from 'papaparse';
+import * as dotenv from 'dotenv';
+
+// Load environment overrides
+dotenv.config();
+
+// Get the current path
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+async function toReturnRecords(queryFN, resultsFNs, options = {}) {
+  // Read the SQL query code.
+  const fullQueryFN = path.resolve(__dirname, queryFN);
+  let sql = null;
+  try {
+    sql = await fs.readFile(fullQueryFN, { encoding: 'utf8', flag: 'r' });
+    sql = sql.trim().replace(/;$/g, '');  // Remove trailing semicolon
+    if (options.orderBy) {
+      // get rid of the order by line for query03
+      if(fullQueryFN.includes('query03')){
+        let lines = sql.split('\n');
+        let lineNumberOfOrderby;
+        let len = lines.length;
+        for(let i = 0; i < len; i++) {
+          if (lines[i].toLowerCase().includes('order by')) {
+            lineNumberOfOrderby = i;
+          }
+        }
+        let sql_new = sql.replace(lines[lineNumberOfOrderby], '');
+        sql = sql_new;
+      }
```

Comment:
**suggestion:** I like this as a start. This was a bug that several students ran into on assignment one (i.e. including `ORDER BY` in their query, and then me doing an explicit `orderBy` in the test).

I was thinking of one of two options to potentially resolve the issue:
1.  Wrap the student's query in an outer query. For example, if the student's SQL is in the `sql` variable, then the SQL that the test would run is:
    ```js
    const wrappedSql = `SELECT * FROM (${sql}) AS query`;
    ```
    Then add `ORDER BY` and `LIMIT` to the `wrappedSql` as needed. Or,
2.  Strip out the `ORDER BY` clause from the students' queries.

Your solution is pretty in line with option 2, except I would suggest instead of just getting rid of the line that has `order by`, get rid of everything after order by too. The reason is that the ordering options need not necessarily be on the same line as the `ORDER BY` keywords (and who knows whether a student would always put them there). According to the [`SELECT` docs](https://www.postgresql.org/docs/current/sql-select.html), the only thing that students might use in their queries that comes after `ORDER BY` is `LIMIT`, so not a big risk.